Content.Server/Atmos/EntitySystems/AutomaticAtmosSystem.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Content.Server.Atmos.Components;
+using Content.Server.Shuttles.EntitySystems;
+using Content.Shared.Maps;
+using Robust.Shared.Map;
+
+namespace Content.Server.Atmos.EntitySystems;
+
+/// <summary>
+/// Handles automatically adding a grid atmosphere to grids that become large enough, allowing players to build shuttles
+/// with a sealed atmosphere from scratch.
+/// </summary>
+public sealed class AutomaticAtmosSystem : EntitySystem
+{
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly ITileDefinitionManager _tileDefinitionManager = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        _mapManager.TileChanged += MapManagerOnTileChanged;
+    }
+
+    private void MapManagerOnTileChanged(object? sender, TileChangedEventArgs e)
+    {
+        // Only if a atmos-holding tile has been added or removed.
+        // Also, these calls are surprisingly slow.
+        // TODO: Make tiledefmanager cache the IsSpace property, and turn this lookup-through-two-interfaces into
+        // TODO: a simple array lookup, as tile IDs are likely contiguous, and there's at most 2^16 possibilities anyway.
+        if (!((e.OldTile.IsSpace(_tileDefinitionManager) && !e.NewTile.IsSpace(_tileDefinitionManager)) ||
+            (!e.OldTile.IsSpace(_tileDefinitionManager) && e.NewTile.IsSpace(_tileDefinitionManager))))
+            return;
```

Comment:
For now, checking `Tile.IsEmpty` before calling `Tile.IsSpace()` might help speed things up a bit. Though maybe IsSpace should just do that automatically?

Reply:
Ideally. When we get diagonal tiles a tile won't necessarily be empty to be space.

Realistically I doubt these calls will show up on a profiler.